Add Plugin: KanboardPermalink

aljawaid opened the pull request:
Add this plugin, also add the following properties to the plugin section:
```
"has_schema": "no",
"has_overrides": "no",
"has_hooks": "yes",
"last_updated": "November 2022"
```
These properties will be called by the (unreleased currently) plugin to show user BEFORE installing which plugins contain potential changes to their Kanboard instance. This has been discussed in the forums as the possible way to resolve the issues with outdated and broken plugins.

Over time, and new plugins will copy the extra properties, but as the properties are optional, we can add them to the docs. Existing plugins will be updated progressively.

aljawaid committed:
FIX: Back2TopBotm Download link

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Add AutomaticActionUX Plugin

aljawaid committed:
Add AutomaticActionUX Plugin

aljawaid committed:
Fix AutomaticActionUX namespace

aljawaid committed:
Add Action Plugin: KanboardEmailHistory

aljawaid committed:
Add Action Plugin: KanboardEmailHistory

Action to email the full history of a task on closure

aljawaid committed:
FIX: Missing comma and download filenames

aljawaid committed:
Merge branch 'main' into master

aljawaid committed:
Released KanboardEmailHistory v1.1

aljawaid committed:
Update plugins.json

aljawaid committed:
Update plugins.json

aljawaid committed:
Merge branch 'main' into master

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Add New Plugin: AgileIndicators

also correct my user handle

aljawaid committed:
Add Plugin: AgileIndicators

also fix namespace for DarkCSS
also correct my user handle

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
New Plugin: KanboardSupport

aljawaid committed:
New Plugin: KanboardSupport

Add a support section in the Kanboard Settings interface so that end users can easily gather any information required by their internal technical support departments and for troubleshooting purposes.

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Update Plugin: KanboardSupport

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Update KanboardSupport Plugin

aljawaid committed:
Update AutomaticActionUX to v1.5

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Update AutomaticActionUX to v1.6

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Add New Plugin: Glancer

aljawaid committed:
Merge branch 'kanboard:main' into master

aljawaid committed:
Add Plugin KanboardPermalink

fguillot commented on plugins.json:
```diff
@@ -1390,5 +1390,21 @@
+        "version": "1.0.0",
+        "author": "Theobald Software GmbH, aljawaid",
+        "license": "MIT",
+        "description": "Glancer lets you quickly jump to a project, task or comment through a search bar at the bottom of the screen. Following the Getting Things Done methodology, productive users can glance in and out of tasks, comments and projects across the interface.",
```
The description seems copy/pasted from the one above

aljawaid replied:
oh damn, sorry I will update the description

aljawaid replied:
it is done

aljawaid committed:
Fix description for KanboardPermalink

fguillot commented on plugins.json:
```diff
@@ -1390,5 +1390,21 @@
+        "has_schema": "no",
+        "has_overrides": "no",
+        "has_hooks": "yes",
+        "last_updated": "November 2022"
```
if you want to add new fields, it would be better to use better types. For example, use boolean instead of a string.

```json
 "has_schema": false, // boolean
 "has_overrides": false,
 "has_hooks": true,
 "last_updated": "2022-11-09" // ISO-8601 date
```

aljawaid replied:
ok I will push another fix

aljawaid committed:
Fix: Change Extra JSON Property Types

- Boolean & ISO-8601 Date type

aljawaid committed:
Update README.md

- Update instructions with links to make it easier for accurate plugin submission by developers
- Add optional property types
- Specify mandatory property types
- Add code example templates

aljawaid commented:
@fguillot any update on this
